extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/ProjectionForField.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.hibernate.orm.panache;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Define a field's path for the SELECT statement
+ * It supports the "dot" notation for fields in referenced entities:
+ * the name is composed of the property name for the relationship, followed by a dot ("."), followed by the name of the field.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.PARAMETER)
+public @interface ProjectionForField {
```

Comment:
Maybe `ProjectedFieldName` would be a better name, it reads more nicely and still carry the projection thematic

Reply:
Yep, sure, implemented